### ISSUE TITLE
Answers are not saved every time when using the button 'Next'

### ISSUE THREAD

**Issue by Kathrin84** [cleblond/moodle-qtype_easyonamejs#7]
Hi Carl,

we're using your question type in our Moodle platform. Now I'm testing it with Moodle version 3.1.
I've noticed that answers are not always saved when using the 'Next' button.
When using the 'Previous' button or by directly clicking into another question led to the saving of the answer.

I used the web inspector and detected that the event
moodle-mod_quiz-autosave: Detected a value change in element q84690:1_answer
is NOT FIRED when using the 'Next' button. 
I think this is the reason why the answer then is not saved.

Could you please have a look at it.
If wanted, I can attach a screencast.

Cheers,
Kathrin

**Comment by kaitkrull** [comment]
Hi Kathrin,

Thank you for describing this problem. We also encountered this issue on our Moodle and it turned out that the problem was in module.js:95 (insert_easyonamejs_applet: function) where necessary functionality was assigned only for one button. So if no back and forward navigation is allowed, then everything works as planned, but if there is one more button, then no answers will be saved if pressed on "Next" button. Answers will be saved if pressed on "Previous" button.

So, the fix itself is simple (line 95 in module.js):
`var nextbutton = inputform.one('input[type=submit]');`
to
`var nextbutton = inputform.all('input[type=submit]');`

Also, during development I found that the path for promise-0.1.1.min.js in edit_easyonamejs_form.php is wrong and should be fixes like it is done in renderer.php.

All the best,
Kait
